internal/domain/file_content_response.go

```diff
@@ -0,0 +1,9 @@
+package domain
+
+type FileContentResponse struct {
+	Filename  string `json:"filename"`
+	Content   string `json:"content"`
+	StartLine int    `json:"start_line"`
+	EndLine   int    `json:"end_line"`
```

Comment:
Consider adding validation logic to verify that StartLine is less than EndLine, and that Position is within valid bounds. This will prevent logical errors when using this struct.